documentation/github-workflows-reference.adoc

```diff
@@ -0,0 +1,47 @@
+:toc:
+toc::[]
+
+== GitHub Actions workflow reference
+
+Currently, the GitHub action workflow is a combination of iterating over several GitHub runners for the creation of OS/architecture specific native images and the usage of the Maven Assembly Plugin to pack the documentation and binaries into their respective release archives.
+
+We have two main workflow files:
+
+== nightly-build.yml
+
+For the creation of SNAPSHOT releases on our OSS Nexus using two steps:
+
+=== build-natives
+
+The nightly build iterates over each OS/architecture runner e.g. `windows-latest` (Windows x64), `macos-latest` (Mac ARM) etc. and builds a binary image.
+Then the job uploads the binaries to GitHub as an artifact using the https://github.com/actions/upload-artifact[GitHub `actions-upload`] tool.
```

Comment:
A small adjustment on my side would be to link `actions-upload` without Github, as that would look better in the docs